src/sspmprk.jl

```diff
@@ -0,0 +1,924 @@
+    end
+
+    # solve linear system
+    linprob = LinearProblem(M, rhs)
+    sol = solve(linprob, alg.linsolve)
+    u2 = sol.u
+    u = u2
+    integrator.stats.nsolve += 1
+
+    # compute Patankar weight denominator
+    ρ = n1 * u + n2 * u .^ 2 ./ σ
+    # avoid division by zero due to zero Patankar weights
+    ρ = add_small_constant(ρ, small_constant)
+
+    P2 = f.p(u, p, t + β10 * dt)
+    Ptmp = β20 * P + β21 * P2
+    integrator.stats.nf += 1
+
+    # build linear system matrix and rhs
+    if f isa PDSFunction
+        d2 = f.d(u, p, t + β10 * dt)  # evaluate nonconservative destruction terms
+        dtmp = β20 * d + β21 * d2
+        rhs = α20 * uprev + α21 * u2 + dt * diag(Ptmp)
+        M = build_mprk_matrix(Ptmp, ρ, dt, dtmp)
+
+    else
+        rhs = α20 * uprev + α21 * u2
+        M = build_mprk_matrix(Ptmp, ρ, dt)
+    end
+
+    # solve linear system
+    linprob = LinearProblem(M, rhs)
+    sol = solve(linprob, alg.linsolve)
+    u = sol.u
+    integrator.stats.nsolve += 1
+
+    # compute Patankar weight denominator
+    σ = σ .^ (1 - s) .* u2 .^ s
+    # avoid division by zero due to zero Patankar weights
+    σ = add_small_constant(σ, small_constant)
+
+    Ptmp = η3 * P + η4 * P2
+
+    # build linear system matrix and rhs
+    if f isa PDSFunction
+        dtmp = η3 * d + η4 * d2
+
+        # see (3.25 f) in original paper
+        rhs = η1 * uprev + η2 * u2 + dt * (η5 * diag(P) + η6 * diag(P2))
+
+        M = build_mprk_matrix(Ptmp, σ, dt, dtmp)
+    else
+        rhs = η1 * uprev + η2 * u2
+        M = build_mprk_matrix(Ptmp, σ, dt)
+    end
+
+    # solve linear system
+    linprob = LinearProblem(M, rhs)
+    sol = solve(linprob, alg.linsolve)
+    σ = sol.u
+    integrator.stats.nsolve += 1
+
+    # compute Patankar weight denominator
+    σ = σ + z .* uprev .* u ./ ρ
+    # avoid division by zero due to zero Patankar weights
+    σ = add_small_constant(σ, small_constant)
+
+    P3 = f.p(u, p, t + c3 * dt)
+    Ptmp = β30 * P + β31 * P2 + β32 * P3
+    integrator.stats.nf += 1
+
+    # build linear system matrix
+    if f isa PDSFunction
+        d3 = f.d(u, p, t + c3 * dt)  # evaluate nonconservative destruction terms
+        dtmp = β30 * d + β31 * d2 + β32 * d3
+        rhs = α30 * uprev + α31 * u2 + α32 * u + dt * diag(Ptmp)
+        M = build_mprk_matrix(Ptmp, σ, dt, dtmp)
+    else
+        rhs = α30 * uprev + α31 * u2 + α32 * u
+        M = build_mprk_matrix(Ptmp, σ, dt)
+    end
+
+    # solve linear system
+    linprob = LinearProblem(M, rhs)
+    sol = solve(linprob, alg.linsolve)
+    u = sol.u
+    integrator.stats.nsolve += 1
+
+    #TODO: Figure out if a second order approximation of the solution
+    # is hidden somewhere.
+    #=
+    tmp = u - σ
+    atmp = calculate_residuals(tmp, uprev, u, integrator.opts.abstol,
+                               integrator.opts.reltol, integrator.opts.internalnorm, t)
+    integrator.EEst = integrator.opts.internalnorm(atmp, t)
+    =#
```

Comment:
Let's come back to this: since this is not implemented so far, the method cannot be adaptive, can it?

- Do we have tests with adaptivity?
- Shall we set `isadaptive(...) = false` for these methods?

Similar questions hold for the other third-order SSPMPRK methods.

Reply:
- The test for different linear solvers uses adaptivity (for adaptive schemes). The tests for different matrix types do not, although they could. I will change this.
- I don't know if setting `isadaptive` is helpful in some situations. `SSPMPRK43` is a subtype of `OrdinaryDiffEqAlgorithm`, not of `OrdinaryDiffEqAdaptiveAlgorithm`. That's how `OrdinaryDiffEq` knows that it is not an adatpive scheme.
- There is only one third order SSPMPRK scheme.

Reply:
I have an idea how to compute a first order approximation within SSPMPRK22, which can be used to estimate the error. But I would like to include this in a future PR.

Reply:
Thanks a lot, this is great 👍 

> There is only one third order SSPMPRK scheme.

Sorry, I wasn't clear - by "method", I meant a Julia method, i.e., the other implementations of the math algorithm for different setups (in-place vs. out-of-place, conservative vs. nonconservative).